### PR TITLE
Update compiler support now that Ubuntu Xenial is out of support

### PR DESCRIPTION
**Pull request by jbytheway**
#### Summary
Infrastructure "Update compiler support now that Ubuntu Xenial is out of support"

#### Purpose of change
Our oldest supported compilers are dictated by the oldest currently supported Ubuntu LTS release.  A couple of months ago Xenial went out of general support and now Bionic is the oldest.  (Xenial still has security support, but that's not sufficient reason to hold ourselves back).

#### Describe the solution
Update the documentation regarding supported compiler versions.

We now target gcc 7.5+ and clang 6+ (and Visual Studio; I'm not changing its version in this PR).

Update Travis CI configuration for the newer versions.

#### Describe alternatives you've considered
We're in a bit of a tricky situation with regard to gcc 7.3.  Bionic defaults to 7.5 on x86, but 7.3 on some other platforms (including ARM).  We probably don't need to worry about Ubuntu LTS on non-x86 platforms, so will just test down to 7.5 for now.  We can always add a 7.3 test later if it proves useful.

#### Testing
You can see the new markdown as rendered by GitHub [here](https://github.com/jbytheway/Cataclysm-DDA/blob/update_compiler_support/doc/COMPILING/COMPILER_SUPPORT.md).

Testing the Travis configuration will require it running on Travis.  Unfortunately Travis recently changed their billing model so no Travis CI is running.  ~It's not currently clear how we will resolve this, so I'm opening this PR as a draft for comment.~  It's very unlikely that my changes to the Travis config are all correct; there are a lot of awkward corner cases.

Update: Kevin suggests not worrying about Travis, so I've expanded the GitHub matrix build to cover this new range of versions.

Update 2: I have deleted the `.travis.yml` since it's probably not correct, but the deleted version will remain in the git history (so long as this PR isn't squashed) so my best guess will be there in case we ever want to support Travis again.

#### Additional context

### CODE REVIEW

**Comment by kevingranade** [comment]
I think you're losing sight of why we're maintaining support for old compiler versions.  It's so that people don't have to upgrade their default compiler to build the game.  ARM is pretty much irrelevant for this as it's an incredibly uncommon build system, we mostly support it as a build target.  To the extent that we want to test on ARM, we mostly just want to be able to run the tests there.

The future of Travis is very uncertain right now, and we actually aren't even triggering it at the moment (something permissions related, maybe). It might be better to focus on the GH actions build matrix.

File: doc/COMPILING/COMPILER_SUPPORT.md [hunk 1]
```diff
@@ -8,25 +8,38 @@ To that end, we aim to support gcc and clang up to the newest stable versions
 and back to those shipping in any supported version of a popular distribution
 or relevant development environment, including Ubuntu, Debian, MSYS, and XCode.
 
+At time of writing:
```
Comment (actual-nh) [inline doc/COMPILING/COMPILER_SUPPORT.md:11]:
```suggestion
At the time of writing:
```

**Comment by jbytheway** [comment]
> I think you're losing sight of why we're maintaining support for old compiler versions. It's so that people don't have to upgrade their default compiler to build the game. ARM is pretty much irrelevant for this as it's an incredibly uncommon build system, we mostly support it as a build target. To the extent that we want to test on ARM, we mostly just want to be able to run the tests there.

I was thinking ARM might be common enough that we care, but if you think otherwise then that makes things easier, so I'll run with that.

> The future of Travis is very uncertain right now, and we actually aren't even triggering it at the moment (something permissions related, maybe). It might be better to focus on the GH actions build matrix.

OK, I'll try to port the Travis coverage over to the GitHub build matrix.

**Comment by jbytheway** [comment]
To provide a little more context, I was worrying about ARM support for something like someone wanting to play on a Raspberry Pi.  However, upon reflection, a platform like that is very unlikely to be running an Ubuntu LTS release, so I am now in agreement that we don't need to worry about it.

**Comment by jbytheway** [comment]
OK, I've updated the OP in light of the move towards GitHub Actions.

I've expanded the matrix job to cover the min and max gcc and clang versions.  I imagine we don't want to test every version on every PR, and (as far as I can tell) there's not an easy way to disable specific jobs for PRs while keeping them for branches.  So I'm not going to worry about testing all the intermediate versions in this PR.  We can set up another workflow to do that on master only if we want.

Just trying to work through the last couple of niggles now.

My biggest remaining question is what to do with `.travis.yml`.  I could leave my edited version in this PR which is something of a step towards a useful version in case we ever re-enable Travis, or I could just delete it.  I'm inclined to just delete it.  Any thoughts?

**Comment by jbytheway** [comment]
(I stole the fixes from #49706 and #49710 to get the CI passing enough to test this PR; credit to @anothersimulacrum)

**Comment by jbytheway** [comment]
Looks like all the CI is green (except Appveyor hasn't finished, but I didn't change that here) so I think this is good to go.

**Comment by actual-nh** [comment]
@jbytheway: Just in case you didn't notice - there's a conflicting file now.

**Comment by jbytheway** [comment]
> @jbytheway: Just in case you didn't notice - there's a conflicting file now.

Thanks. Indeed I hadn't. Rebased.

**Comment by jbytheway** [comment]
Kevin, thanks for fixing those merge conflict markers I missed!

**Pull request merged**